Xamarin.Forms.Build.Tasks/CompiledMarkupExtensions/StaticExtension.cs

```diff
@@ -31,74 +34,104 @@ public IEnumerable<Instruction> ProvideValue(IElementNode node, ModuleDefinition
 			var getterDef = module.ImportReference(propertyDef.GetMethod);
-			return new [] { Instruction.Create(OpCodes.Call, getterDef) };
+			return new[] { Instruction.Create(OpCodes.Call, getterDef) };
 		}
 
 		public static FieldReference GetFieldReference(TypeReference typeRef, string fieldName, ModuleDefinition module)
 		{
-			TypeReference declaringTypeReference;
-			FieldReference fRef = typeRef.GetField(fd => fd.Name == fieldName &&
-			                                       fd.IsStatic &&
-			                                       fd.IsPublic, out declaringTypeReference);
+			FieldReference fRef = typeRef.GetField(fd => fd.Name == fieldName
+														&& fd.IsStatic
+														&& IsPublicOrVisibleInternal(fd, module), out TypeReference declaringTypeReference);
 			if (fRef != null) {
 				fRef = module.ImportReference(fRef.ResolveGenericParameters(declaringTypeReference));
 				fRef.FieldType = module.ImportReference(fRef.FieldType);
 			}
 			return fRef;
 		}
 
+		static bool IsPublicOrVisibleInternal(FieldDefinition fd, ModuleDefinition module)
+		{
+			if (fd.IsPublic)
+				return true;
+			if (fd.IsAssembly) {
+				if (fd.Module == module)
+					return true;
+				if (fd.Module.GetCustomAttributes().Any(ca =>  ca.AttributeType.FullName == "System.Runtime.CompilerServices.InternalsVisibleToAttribute"
+															&& ca.HasConstructorArguments
+															&& (ca.ConstructorArguments[0].Value as string) != null
+															&& (ca.ConstructorArguments[0].Value as string).StartsWith(module.Assembly.Name.Name, System.StringComparison.InvariantCulture)))
+					return true;
+			}
+			return false;
+		}
+
 		public static PropertyDefinition GetPropertyDefinition(TypeReference typeRef, string propertyName, ModuleDefinition module)
 		{
-			TypeReference declaringTypeReference;
-			PropertyDefinition pDef = typeRef.GetProperty(pd => pd.Name == propertyName &&
-			                                              pd.GetMethod.IsPublic &&
-			                                              pd.GetMethod.IsStatic, out declaringTypeReference);
+			PropertyDefinition pDef = typeRef.GetProperty(pd => pd.Name == propertyName
+																&& IsPublicOrVisibleInternal(pd.GetMethod, module)
+																&& pd.GetMethod.IsStatic, out TypeReference declaringTypeReference);
 			return pDef;
 		}
+
+		static bool IsPublicOrVisibleInternal(MethodDefinition md, ModuleDefinition module)
+		{
+			if (md.IsPublic)
+				return true;
+			if (md.IsAssembly) {
+				if (md.Module == module)
+					return true;
+				if (md.Module.GetCustomAttributes().Any(ca => ca.AttributeType.FullName == "System.Runtime.CompilerServices.InternalsVisibleToAttribute"
+															&& ca.HasConstructorArguments
+															&& (ca.ConstructorArguments[0].Value as string) != null
+															&& (ca.ConstructorArguments[0].Value as string).StartsWith(module.Assembly.Name.Name, System.StringComparison.InvariantCulture)))
+					return true;
```

Comment:
`> return md.Module.GetCustomAttributes().Any(...`

Reply:
I like to return `false` from a single point, in case we have to add a 3rd case (I don't know, for `protected` fields)